python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+        deferLoad=True,
+        multiple=True,
+    )
+    inputVisitSummary = pipeBase.connectionTypes.Input(
+        doc=("Per-visit consolidated exposure metadata built from calexps. "
+             "These catalogs use detector id for the id and must be sorted for "
+             "fast lookups of a detector."),
+        name="visitSummary",
+        storageClass="ExposureCatalog",
+        dimensions=("instrument", "visit"),
+        multiple=True,
+    )
+    referenceCatalog = pipeBase.connectionTypes.PrerequisiteInput(
+        doc="The astrometry reference catalog to match to loaded input catalog sources.",
+        name="gaia_dr2_20200414",
+        storageClass="SimpleCatalog",
+        dimensions=("skypix",),
+        deferLoad=True,
+        multiple=True,
+        lookupFunction=lookupVisitRefCats,
+    )
+    outputWcs = pipeBase.connectionTypes.Output(
+        doc=("Per-tract, per-visit world coordinate systems derived from the fitted model."
+             " These catalogs only contain entries for detectors with an output, and use"
+             " the detector id for the catalog id, sorted on id for fast lookups of a detector."),
+        name="WCSFitSkyWcsCatalog",
+        storageClass="ExposureCatalog",
+        dimensions=("instrument", "visit", "skymap", "tract"),
+        multiple=True
+    )
+    outputCatalog = pipeBase.connectionTypes.Output(
+        doc=("Source table with stars used in fit, along with residuals in pixel coordinates and tangent "
+             "plane coordinates and chisq values"),
+        name="WCSFit_usedStars",
+        storageClass="DataFrame",
+        dimensions=("instrument", "skymap", "tract"),
+    )
+
+
+class WCSFitConfig(pipeBase.PipelineTaskConfig,
+                   pipelineConnections=WCSFitConnections):
+    """Configuration for WCSFitTask"""
+    sourceSelector = sourceSelectorRegistry.makeField(
+        doc="How to select sources for cross-matching",
+        default="science"
+    )
+    referenceSelector = pexConfig.ConfigurableField(
+        target=ReferenceSourceSelectorTask,
+        doc="How to down-select the loaded astrometry reference catalog.",
+    )
+    matchRadius = pexConfig.Field(
+        doc="Matching tolerance between associated objects (arcseconds)",
+        dtype=float,
+        default=1.0
+    )
+    minMatches = pexConfig.Field(
+        doc="Number of matches required to keep a source object",
+        dtype=int,
+        default=2
+    )
+    allowSelfMatches = pexConfig.Field(
+        doc="Allow multiple object matches from the same viist",
+        dtype=bool,
+        default=False
+    )
+    sourceFluxType = pexConfig.Field(
+        dtype=str,
+        doc="Source flux field to use in source selection and to get fluxes from the catalog.",
+        default='apFlux_12_0'
+    )
+    systematicError = pexConfig.Field(
+        dtype=float,
+        doc="Systematic error padding for the science images (marcsec)",
```

Comment:
Is this a padding, or something added in quadrature?

Reply:
It's added in quadrature, because it gets converted to covariance. Do you think it should be called something else?